Bot.py

```diff
@@ -137,20 +137,24 @@ def change_target(self, tower_type, tower_position, targets: str | list, delay:
 
         current_target_index = 0
 
+        if "SPIKE" in tower_type:
+            target_order = static.target_order_spike
+        else:
+            target_order = static.target_order_regular
+
+        current_target_index = 0
+
         # for each target in target list
         for i in targets:
-            
-            # Math to calculate the difference between current target index and next target index
-            if "SPIKE" in tower_type:
-                target_diff = abs((static.target_order_spike.index(i)) - current_target_index)
+
+            if target_order.index(i) < current_target_index:
+                target_diff = abs(current_target_index - 3 + target_order.index(i) + 1)
```

Comment:
Where does the -3 and +1 come from?

Reply:
actually I did a mistake with the formula. It should have been `target_diff = abs(current_target_index - 3) + target_order.index(i) + 1`, I already fixed it.

And 3 is the max index (we have 4 targets in the arrays) and +1 because arrays start at 0. I'm not good at explaining it but it works